flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotSharingSlotAllocator.java

```diff
@@ -274,4 +298,68 @@ public Collection<ExecutionVertexID> getContainedExecutionVertices() {
             return containedExecutionVertices;
         }
     }
+
+    private static class SlotSharingGroupMetaInfo {
+
+        private final int minLowerBound;
+        private final int maxUpperBound;
+        private final int maxLowerUpperBoundRange;
+
+        private SlotSharingGroupMetaInfo(
+                int minLowerBound, int maxUpperBound, int maxLowerUpperBoundRange) {
+            this.minLowerBound = minLowerBound;
+            this.maxUpperBound = maxUpperBound;
+            this.maxLowerUpperBoundRange = maxLowerUpperBoundRange;
+        }
+
+        public int getMinLowerBound() {
+            return minLowerBound;
+        }
+
+        public int getMaxUpperBound() {
+            return maxUpperBound;
+        }
+
+        public int getMaxLowerUpperBoundRange() {
+            return maxLowerUpperBoundRange;
+        }
+
+        public static Map<SlotSharingGroupId, SlotSharingGroupMetaInfo> from(
+                Iterable<JobInformation.VertexInformation> vertices) {
+
+            return getPerSlotSharingGroups(
+                    vertices,
+                    vertexInformation ->
+                            new SlotSharingGroupMetaInfo(
+                                    vertexInformation.getMinParallelism(),
+                                    vertexInformation.getParallelism(),
+                                    vertexInformation.getParallelism()
+                                            - vertexInformation.getMinParallelism()),
+                    (metaInfo1, metaInfo2) ->
+                            new SlotSharingGroupMetaInfo(
+                                    Math.min(metaInfo1.getMinLowerBound(), metaInfo2.minLowerBound),
```

Comment:
Hi Schepler!  May you explain here why it's not Math.**max**(metaInfo1.getMinLowerBound(), metaInfo2.minLowerBound) but min here?